File: Chain of Responsibility Pattern/TraineeDoctor.java
package traineedoctor;

import patient.Patient;
import diseaseseverity.DiseaseSeverity;
import handler.Handler;

public class TraineeDoctor extends Handler{
	
	@Override
	public void handlePatient(Patient patient){
		
		if(patient.getDiseaseSeverity() == DiseaseSeverity.LOW){
			System.out.println("Diagnose the Patient " + patient.getName() + " by the Trainee Doctor \n");
		}else{
			System.out.println("Passing the Patient " + patient.getName() + " to Higher Authority from Trainee Doctor");
			this.successor.handlePatient(patient);
		}
		
	}
	
}

File: Design Patterns/Chain of Responsibility Pattern in JavaScript/TraineeDoctor.js
const Handler = require('./Handler');
const DiseaseSeverity = require('./DiseaseSeverity');

class TraineeDoctor extends Handler {

    handlePatient(patient) {

        super.handlePatient(patient);

        if (patient.diseaseSeverity === DiseaseSeverity.LOW) {
            console.log("Diagnose the Patient " + patient.name + " by the Trainee Doctor \n");
        } else {
            console.log("Passing the Patient " + patient.name + " to Higher Authority from Trainee Doctor");
            this.successor.handlePatient(patient);
        }

    }

}

module.exports = TraineeDoctor;